syntax = "proto3";
option java_multiple_files = true;

// python -m grpc_tools.protoc -I=proto --python_out=. --grpc_python_out=.
// proto/analyser.proto

package tibava.analyser;

service Analyser {
  rpc copy_data(stream DataRequest) returns (DataResponse);
  rpc list_plugins(ListPluginsRequest) returns (ListPluginsReply) {}
}

enum DataType {
  UNKOWN_TYPE = 0;
  STRING_TYPE = 1;
  INT_TYPE = 2;
  FLOAT_TYPE = 3;
}

message PluginParameter {
  string name = 1;
  string default = 2;
  DataType type = 3;
}

enum PluginDataType {
  UNKOWN_DATA = 0;
  VIDEO_DATA = 1;
  IMAGE_DATA = 2;
}

message PluginData {
  string name = 1;
  PluginDataType type = 2;
}

message PluginInfo {
  string name = 1;
  string version = 2;
  repeated PluginParameter parameters = 3;
  repeated PluginData requires = 4;
  repeated PluginData provides = 5;
}

message ListPluginsRequest {}
message ListPluginsReply { repeated PluginInfo plugins = 1; }

message DataRequest {
  bytes data_encoded = 1;
  PluginDataType type = 2;
}

message DataResponse {
  string id = 1;
}